### PR TITLE
Modified the throw context example with concrete types but not concrete values.

### PR DESCRIPTION
**Pull request by ioannad**
Addresses @aheejin's request in a previous review comment:
https://github.com/WebAssembly/exception-handling/pull/180#discussion_r955477590

Originally I wrote this using
- `val_{i32} = (i32.const 1)`,
- `val_{f32} = (f32.const 2.0)`, and
- `val_{i64} = (i64.const 3)`,

but the example seemed then really long.
To keep the example relatively short I switched to the current version.

### CODE REVIEW

**Comment by ioannad** [comment]
@aheejin is the example clear enough now or do you think it's better to use the concrete values, so `(i32.const 1)` instead of `val_{i32}` etc?

**Comment by ioannad** [comment]
@rossberg, incidentally, this also fixes [your previous comment](https://github.com/WebAssembly/exception-handling/pull/180#discussion_r955923760) about using "We ...".

**Comment by ioannad** [comment]
@rossberg incidentally, this also addresses your [previous comment](https://github.com/WebAssembly/exception-handling/pull/180#discussion_r955919370) about using "let ... be .. ".

File: document/core/exec/runtime.rst [hunk 1]
```diff
@@ -804,7 +805,7 @@ Finally, the following definition of *evaluation context* and associated structu
    \begin{array}{llll}
    \production{(evaluation contexts)} & E &::=&
      [\_] ~|~
-     \val^\ast~E~\instr^\ast ~|~
+     val^\ast~E~\instr^\ast ~|~
```
Comment (aheejin) [inline document/core/exec/runtime.rst]: Why remove `\` here?

File: document/core/exec/runtime.rst [hunk 2]
```diff
@@ -732,31 +732,32 @@ Throw contexts allow matching the program context around a throw instruction up
...
-   When a throw of the form :math:`val^m (\THROWadm~a)` occurs, we search for the maximal surrounding throw context :math:`T`,
-   which means we pop any other values, labels, frames, and |CAUGHTadm| instructions surrounding the throw :math:`val^m (\THROWadm~a)`,
+   When a throw of the form :math:`\val^m (\THROWadm~a)` occurs, we search for the maximal surrounding throw context :math:`T`,
+   which means we pop any other values, labels, frames, and |CAUGHTadm| instructions surrounding the throw :math:`\val^m (\THROWadm~a)`,
    until we find an exception handler (corresponding to a try block) that :ref:`handles the exception <syntax-handler>`.
-   We then append the values :math:`val^m:[t^m]` to the tag address :math:`a` into a new |CAUGHTadm| instruction which we push on the stack.
+   We then append the values :math:`\val^m:[t^m]` to the tag address :math:`a` into a new |CAUGHTadm| instruction which we push on the stack.
 
    In other words, when a throw occurs, normal execution halts and exceptional execution begins, until the throw
    is the continuation (i.e., in the place of a :math:`\_`) of a throw context in a catching try block.
```
Comment (aheejin) [inline document/core/exec/runtime.rst]: Not strictly related to this PR, but I'm not sure if I understand this sentence.
- What does a hole represent? (I know this is a basic question, but "block context" says it is "the next step of computation is taking place".. Is that also the case here? Which next step are we talking about?)
- This sentence sounds like we have two separate moments of "exception execution begins" and "(until) the throw is the continuation". I thought the exception execution begins when the `throw` is executed... Are they two different things?

**Review by aheejin — APPROVED** [review]
LGTM % comments. The concrete example is a lot easier to understand. Thanks!

File: document/core/exec/runtime.rst (hunk 2 above)
Comment (aheejin) [reply document/core/exec/runtime.rst]: By the way this is a general question, so please don't consider this question as a blocker for landing this PR.

File: document/core/exec/runtime.rst (hunk 1 above)
Comment (ioannad) [reply document/core/exec/runtime.rst]: I removed `\` from all the `var_{...}` in the example because there `var` is part of a variable name (could have used anything, like `x`, `y`, `z` ), as opposed to `\var`which is a  [non-terminal symbol](https://webassembly.github.io/spec/core/syntax/conventions.html#grammar-notation). However using the non-terminal symbols as variable names is widely used in the spec, so I guess it might be confusing to change in this example. Putting all the `\` back before merging.

File: document/core/exec/runtime.rst (hunk 2 above)
Comment (ioannad) [reply document/core/exec/runtime.rst]: > This sentence sounds like we have two separate moments of "exception execution begins" and "(until) the throw is the continuation". I thought the exception execution begins when the throw is executed... Are they two different things?

@aheejin this is exactly what I meant, so by "exceptional execution" I meant the execution of the throw, so the popping of values and labels until the first catching try block is reached. I was only trying to describe the execution of a throw instruction, but I see now I overcomplicated it, making it seem as if there are more than one execution steps during "exceptional execution". 

About the hole, I think of the hole as the position of the stack pointer of the Wasm stack machine, but I see how mentioning the hole here can be confusing as well. Thank you for  pointing this out! 

Rephrasing these two lines to:

>    In other words, a thrown exception is caught when it's the continuation of a throw context in a catching try block, i.e., it's inside a catching try block, which is the innermost with respect to the throw. 

Is it clearer like this?

Comment (aheejin) [reply document/core/exec/runtime.rst]: > when it's the continuation of a throw context in a catching try block, 

What's 'it' here? It looks to be "a thrown exception" looking at the first part of the sentence, but I'm not sure what it means that an exception is a continuation of a throw context.

File: document/core/exec/runtime.rst [hunk 3]
```diff
@@ -732,34 +732,35 @@ Throw contexts allow matching the program context around a throw instruction up
...
+   :math:`T=[\val_{i32}\_]`, with the exception handler :math:`H=\CATCHadm\{a~\epsilon\}` gives:
 
    .. math::
       \begin{array}{lll}
-      \stepto & S;~F;~\LABEL_m\{\}~(\CAUGHTadm\{a~\val^m\}~\val^m~\RETURN~\END)~\END & \hspace{9ex}\ \\
-      \stepto & \val^m & \\
+      \stepto & F;~\LABEL_2\{\}~(\CAUGHTadm\{a~\val_{f32}~\val_{i64}\}~\val_{f32}~\val_{i64}~\END)~\END & \hspace{9ex}\ \\
+      \stepto & F;~\LABEL_2\{\}~\val_{f32}~\val_{i64}~\END & \hspace{9ex}\ \\
+      \stepto & \val_{f32}~\val_{i64} & \\
       \end{array}
```
Comment (aheejin) [inline document/core/exec/runtime.rst]: How many newlines do we usually add between paragraphs? Here we have three... Is that normal?

File: document/core/exec/runtime.rst [hunk 4]
```diff
@@ -732,34 +732,35 @@ Throw contexts allow matching the program context around a throw instruction up
 .. note::
    For example, catching a simple :ref:`throw <exec-throw>` in a :ref:`try block <exec-try-catch>` would be as follows.
```
Comment (rossberg) [inline document/core/exec/runtime.rst]:
```suggestion
   For example, catching a simple :ref:`throw <syntax-throw>` in a :ref:`try block <syntax-try-catch>` would be as follows.
```

File: document/core/exec/runtime.rst [hunk 5]
```diff
@@ -732,34 +732,35 @@ Throw contexts allow matching the program context around a throw instruction up
...
 
-   Assume that :math:`\expand_F(bt) = [t1^n] \to [t2^m]`, for some :math:`n > m` such that :math:`t1^n[(n-m):n] = t2^m`,
-   and that the tag address `a` of :math:`x` corresponds to the tag type :math:`[t2^m] \to []`.
+   Assume that :math:`\expand_F(bt) = [\I32~\F32~\I64] \to [\F32~\I64]`,
+   and that the tag address `a` of :math:`x` corresponds to the tag type :math:`[\F32~\I64] \to []`.
```
Comment (rossberg) [inline document/core/exec/runtime.rst]:
```suggestion
   and that the tag address `a` of :math:`x` has tag type :math:`[\F32~\I64] \to []`.
```

File: document/core/exec/runtime.rst [hunk 6]
```diff
@@ -732,34 +732,35 @@ Throw contexts allow matching the program context around a throw instruction up
...
+      \stepto & F;~\LABEL_2\{\}~(\CAUGHTadm\{a~\val_{f32}~\val_{i64}\}~\val_{f32}~\val_{i64}~\END)~\END & \hspace{9ex}\ \\
+      \stepto & F;~\LABEL_2\{\}~\val_{f32}~\val_{i64}~\END & \hspace{9ex}\ \\
+      \stepto & \val_{f32}~\val_{i64} & \\
       \end{array}
 
 
 
-   When a throw of the form :math:`val^m (\THROWadm~a)` occurs, we search for the maximal surrounding throw context :math:`T`,
-   which means we pop any other values, labels, frames, and |CAUGHTadm| instructions surrounding the throw :math:`val^m (\THROWadm~a)`,
+   When a throw of the form :math:`\val^m (\THROWadm~a)` occurs, we search for the maximal surrounding throw context :math:`T`,
```
Comment (rossberg) [inline document/core/exec/runtime.rst]: Avoid "we" :)

File: document/core/exec/runtime.rst [hunk 7]
```diff
@@ -732,34 +732,35 @@ Throw contexts allow matching the program context around a throw instruction up
...
+   which means we pop any other values, labels, frames, and |CAUGHTadm| instructions surrounding the throw :math:`\val^m (\THROWadm~a)`,
    until we find an exception handler (corresponding to a try block) that :ref:`handles the exception <syntax-handler>`.
-   We then append the values :math:`val^m:[t^m]` to the tag address :math:`a` into a new |CAUGHTadm| instruction which we push on the stack.
+   We then append the values :math:`\val^m:[t^m]` to the tag address :math:`a` into a new |CAUGHTadm| instruction which we push on the stack.
 
-   In other words, when a throw occurs, normal execution halts and exceptional execution begins, until the throw
-   is the continuation (i.e., in the place of a :math:`\_`) of a throw context in a catching try block.
+   In other words, a thrown exception is caught when it's the continuation of a throw context in a catching try block,
```
Comment (rossberg) [inline document/core/exec/runtime.rst]: I'm afraid I don't understand this sentence. How can an exception be a continuation?

File: document/core/exec/runtime.rst [hunk 8]
```diff
@@ -732,34 +732,35 @@ Throw contexts allow matching the program context around a throw instruction up
...
+   i.e., it's inside a catching try block, which is the innermost with respect to the throw.
```
Comment (rossberg) [inline document/core/exec/runtime.rst]:
```suggestion
   i.e., it is inside a catching try block, which is the innermost with respect to the throw.
```

File: document/core/exec/runtime.rst (hunk 1 above)
Comment (rossberg) [reply document/core/exec/runtime.rst]: Writing "var" in TeX math mode is the same as "v a r", i.e., three variables. among other things, the macros ensure that it's correctly type-set as "\mathit{var}".

**Review by rossberg — CHANGES_REQUESTED** [review]

File: document/core/exec/runtime.rst (hunk 2 above)
Comment (ioannad) [reply document/core/exec/runtime.rst]: By "the thrown exception" I meant the instruction that throws the exception,  i.e., `val* (throw a)`.

I guess it's better to call it "the exception throw" instead, reformulating.

File: document/core/exec/runtime.rst (hunk 3 above)
Comment (ioannad) [reply document/core/exec/runtime.rst]: Thanks for catching this typo!

File: document/core/exec/runtime.rst (hunk 6 above)
Comment (ioannad) [reply document/core/exec/runtime.rst]: Oops! :see_no_evil: fixing.

File: document/core/exec/runtime.rst (hunk 2 above)
Comment (ioannad) [reply document/core/exec/runtime.rst]: @aheejin I just realised this is all superseded by your PR #232 so leaving this for now.

File: document/core/exec/runtime.rst (hunk 6 above)
Comment (ioannad) [reply document/core/exec/runtime.rst:757]: I just realised this part is superseded by @aheejin 's PR #232, so leaving this as is for now.

Comment (ioannad) [reply document/core/exec/runtime.rst]: This is fixed by @aheejin's PR #232 :)

File: document/core/exec/runtime.rst (hunk 7 above)
Comment (ioannad) [reply document/core/exec/runtime.rst]: @aheejin had the [same remark](https://github.com/WebAssembly/exception-handling/pull/219#discussion_r972985641) and I think you are both right that this doesn't make sense. Rephrasing "the thrown exception" to "the exception throw". Is it better like this?

Comment (ioannad) [reply document/core/exec/runtime.rst]: (done)

File: document/core/exec/runtime.rst (hunk 2 above)
Comment (ioannad) [reply document/core/exec/runtime.rst]: (done)

File: document/core/exec/runtime.rst (hunk 7 above)
Comment (rossberg) [reply document/core/exec/runtime.rst]: I'm afraid I still don't understand the bit about the exception being the continuation of a context.

File: document/core/exec/runtime.rst (hunk 6 above)
Comment (ioannad) [reply document/core/exec/runtime.rst]: Now that #232 landed, I merged the current main branch, in which this is fixed.

File: document/core/exec/runtime.rst (hunk 7 above)
Comment (ioannad) [reply document/core/exec/runtime.rst]: To be honest, I really can't remember anymore why this is written like this in the first place (2nd EH proposal), and reading this again I agree it doesn't make much sense, so I removed this line in the last commit.

File: document/core/exec/runtime.rst (hunk 1 above)
Comment (ioannad) [reply document/core/exec/runtime.rst]: Tihs suggestion is applied in the last commit.

File: document/core/exec/runtime.rst (hunk 2 above)
Comment (ioannad) [reply document/core/exec/runtime.rst]: This is now entirely removed, after [this discussion](https://github.com/WebAssembly/exception-handling/pull/219#discussion_r992420960).

**Comment by ioannad** [comment]
All review comments should be addressed now, pending for approval.

File: document/core/exec/runtime.rst (hunk 7 above)
Comment (ioannad) [reply document/core/exec/runtime.rst]: This sentence mentioning continuations is now entirely removed.

**Review by rossberg — APPROVED** [review]
